Passage à Django 1.8

GerardPaligot opened the pull request:
Suite de #3245 avec un rebase et plus propre :D

| Q | R |
| --- | --- |
| Correction de bugs ? | Non |
| Nouvelle Fonctionnalité ? | Oui |
| Tickets (_issues_) concernés | #2507, https://github.com/zestedesavoir/zds-site/issues/2666 |
# QA

Tester tout ! :)
# Pourquoi cette PR ?

Oui, pourquoi cette PR alors que @gustavi est en train de [développer la compatibilité Django 1.9](https://github.com/zestedesavoir/zds-site/pull/3245) ? Tout simplement parce que j'estime que le projet n'est pas encore prêt pour cette version de Django.

La situation actuelle est la suivante :
- Django 1.8 est la dernière version LTS de Django. Cela ne veut pas dire que nous y resterons longtemps, cela veut simplement dire que cette version est stable et qu'elle bénéficiera des correctifs de sécurité pendant 3 ans. Après, libre à nous de rester 3 ans ou d'évoluer rapidement vers Django 1.9.
- Les dépendances de nos projets ne sont pas encore compatibles avec Django 1.9 et tardent à le faire. Puisque nous n'utiliserons et ne testerons pas une PR qui comportent des dépendances non compatibles, passer à la version 1.8 qui est déjà compatible avec toutes les dépendances me semble un choix plus que judicieux.

Pour ces 2 raisons, j'ai repris le travail de @gustavi et j'ai rollback à la version 1.8.

Commit:
Model._meta.module_name --> Model._meta.model_name

Cf https://docs.djangoproject.com/en/1.8/releases/1.8/#features-removed-in-1-8

Commit:
Suppression du module inutilisé "django.contrib.sites"

S'il doit revenir, des modifications sont nécessaires.
Cf https://docs.djangoproject.com/en/1.8/releases/1.8/#django-contrib-sites

Commit:
Django 1.8: Déclarer le dossier de fixtures par défaut lance une erreur

Donc on supprime cette déclaration qui était de toutes façons inutile même avec Django 1.7.
Cf https://docs.djangoproject.com/en/1.8/releases/1.8/#management-commands

Commit:
Suppression d'une relation erronée

Django 1.8 vérifie ces relations et donc lancerait une erreur.
Cf https://docs.djangoproject.com/en/1.8/releases/1.8/#select-related-now-checks-given-fields

Commit:
Utilisation d'une API privée qui a changée de package
Cf https://docs.djangoproject.com/en/1.8/releases/1.8/#cleanup-of-the-django-template-namespace

Commit:
Passage à Django 1.8 minimal (il reste plein de warnings mais ça marche)

Impacts directs obligatoires :
- Une relation en base change de nom pour être python-compatible (et donc sa migration en base)
- On doit maintenant préciser le paramètre --fake-initial lors d'une migration de MAJ (cf https://docs.djangoproject.com/en/1.8/ref/django-admin/#django-admin-option---fake-initial)
- Debug Toolbar en v1.3
- Fix pour que oauth fonctionne, cf https://github.com/evonove/django-oauth-toolkit/issues/204

Commit:
Une relation N..M nullable n'a pas de sens

Ce qui n'a pas de sens doit disparaître, d'autant plus que ça provoque des warnings.
Cf https://docs.djangoproject.com/en/1.8/ref/models/fields/#django.db.models.ManyToManyField.allow_unsaved_instance_assignment

Commit:
RedirectView.permanent va changer de valeur --> on la fixe

Cf https://docs.djangoproject.com/en/1.8/ref/class-based-views/base/#django.views.generic.base.RedirectView.permanent

Commit:
django.utils.tzinfo est obsolète, remplacé par django.utils.timezone

Cf https://docs.djangoproject.com/fr/1.8/ref/utils/#django-utils-tzinfo

artragis committed:
refactor(migration): Supprime les migrations des vieux modules.

artragis committed:
chore(dep): Mise a jour de django-crispy-forms de 1.5.2 a 1.6.0.

artragis committed:
chore(dep): Mise a jour de django de 1.7.10 a 1.8.9.

gustavi committed:
chore(dep): Mise a jour de django-debug-toolbar de 1.3.0 a 1.4.

gustavi committed:
chore(dep): Mise a jour de django-oauth-toolkit de 0.9.0 a 0.10.0.

gustavi committed:
refactor(url): Utilisation nouvelle syntaxe dans les URLs.

- Change le format des urlpatterns.
- Utilise les vues dans la declaration des URLs.
- Correction de tous les reverse.
- Correction de toutes les balises url dans les templates.

gustavi committed:
refactor(1.8): Changement de l'import csrf.

https://docs.djangoproject.com/fr/1.8/ref/templates/api/#django.template.RequestContext

gustavi committed:
refactor(settings): Change la gestion des templates.

gustavi committed:
Correction d'un import

artragis committed:
corrige les erreurs de tutorialv2 et munin

artragis committed:
règle la migration et les avertissements de template du forum

artragis committed:
refactor(form): Retire les warnings sur les forms.

gustavi committed:
chore(dep): Mise a jour de django-munin de 0.2.0 a 0.2.1.

gustavi committed:
Regle le soucis pour Travis et MySQL (merci @DevHugo)

gustavi committed:
Ajout des informations pour la mise en prod de la v17

vhf commented:
:100:

DevHugo commented on doc/source/index.rst:
```diff
@@ -2,7 +2,7 @@
 ================================
 
-Zeste de Savoir est un site internet communautaire codé à l'aide du framework Django 1.7 et de Python 2.7
+Zeste de Savoir est un site internet communautaire codé à l'aide du framework Django 1.9 et de Python 2.7
```
C'est de la publicité mensongère ! ^^

GerardPaligot replied:
Ouais, j'ai vu après des petits détails dans la doc pour Django 1.9. J'attends d'abord de voir si on merge ma PR ou celle de gustavi.

gustavi commented:
Je pense qu'au lieu de perdre du temps là dessus on devrait essayer de fixer les 2 erreurs Travis de Django Haystack.

GerardPaligot commented:
Je n'ai pas l'impression de perdre du temps quand Haystack n'est pas la seule dépendance à poser problème et que le choix de migrer vers Django 1.9 n'a pas été discutée face à une LTS. :)

gustavi commented:
À part Django Haystack qu'est-ce qui ne fonctionne pas ?

vhf commented:
@gustavi Ce n'est pas qu'une question de ce qui fonctionne ou pas.

Que fait-on si on a un gros problème avec la 1.9 qu'on aurait pas remarqué lors de la beta ? Un problème de performance par exemple. Ou de sécurité. Bref un truc très embêtant. On rollback vers la 1.7 qui n'est plus maintenue ? On attend la 2.0 sachant qu'il faudra à nouveau adapter du code à la 2.0 ET attendre que les dépendances suivent ?

Peu importe à quelle vitesse on passe à la 1.9 _après_ la 1.8 et peu importe si on passe à la 1.9 ou si on garde la LTS, passer _d'abord_ à la LTS est un sage décision de mon point de vue.

GerardPaligot commented:
Je ne dirais pas mieux que @vhf mais pour répondre à ta question : haystack, drf-extensions et (anciennement il semblerait) swagger.

gustavi commented:
J'ai envie de dire que c'est la même chose avec Django 1.8.

Après je suis favorable au _LTS_ mais pas pour un projet qui bouge autant que ça. Sur un site perso où je fais 10 commits par an OK, sur un gros projet comme ça je suis pour se mettre à jour petit à petit parce que quand tu dois passer d'une version _n_ à une _n+1_ c'est déjà du gros travail alors de passer à du _n_ à _n+3_ c'est vraiment monstrueux.

@GerardPaligot : swagger est OK.

vhf commented:
> J'ai envie de dire que c'est la même chose avec Django 1.8.

Non. Si on a un problème du genre il sera fixé sur la 1.8. Sur la 1.7 absolument pas, sur la 1.9 on ne peut pas savoir.

Cette PR n'est probablement pas le bon endroit pour débattre de rester sur la 1.8 ou pas. Il est pour débattre de l'utilité de passer à la 1.8 et ceci au risque de me répéter _peu importe si on passe à la 1.9 ou si on garde la LTS_.

gustavi commented:
On a vraiment besoin de QA là dessus

GerardPaligot commented:
Une QA directe sur la bêta me semble plus pertinente (une fois que la release actuelle sera faite).

artragis commented:
En attendant http://zds.francoisdambrine.me vous attend.

pierre-24 commented:
Hey, les identifiants par défauts fonctionnent pas, charge les fixtures ;)

gustavi commented:
@artragis sinon pas de .htaccess ?

artragis commented:
Ça me soulerait d'en mettre un.

Je charge les fixtures @pierre-24

2016-02-01 12:01 GMT+01:00 Laville Augustin notifications@github.com:

> @artragis https://github.com/artragis sinon pas de .htaccess ?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/zestedesavoir/zds-site/pull/3331#issuecomment-177915963
> .

artragis commented:
Les comptes basiques sont ajoutés et j'ai lancé l'ajout des factories high

2016-02-01 12:59 GMT+01:00 francois dambrine artragis@gmail.com:

> Ça me soulerait d'en mettre un.
> 
> Je charge les fixtures @pierre-24
> 
> 2016-02-01 12:01 GMT+01:00 Laville Augustin notifications@github.com:
> 
> > @artragis https://github.com/artragis sinon pas de .htaccess ?
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/zestedesavoir/zds-site/pull/3331#issuecomment-177915963
> > .

firm1 commented:
J'ai noté un bug sur les modales qui prennent tout l'écran.

Par exemple la modale de signalement d'un message dans le forum.

![globule](https://cloud.githubusercontent.com/assets/6066015/12717663/8918e192-c8e9-11e5-8d29-03dd0ed1fb49.PNG)

DevHugo commented:
Pareil. pour ajouter un utilisateur dans un tutoriel.

J'ai joué un peu et j'ai rien vu à part ça.

artragis commented:
J'avais oublié que sur mon serveur, npm n'arrivait pas à builder ça
correctement. Je vais voir plus tard.

2016-02-01 13:47 GMT+01:00 Hugo Courtecuisse notifications@github.com:

> Pareil; pour ajouter un utilisateur dans un tutoriel
> 
> 2016-02-01 13:41 GMT+01:00 firm1 notifications@github.com:
> 
> > J'ai noté un bug sur les modale qui prennent tout l'écran.
> > 
> > Par exemple la modale de signalement d'un message dans le forum.
> > 
> > [image: globule]
> > <
> > https://cloud.githubusercontent.com/assets/6066015/12717663/8918e192-c8e9-11e5-8d29-03dd0ed1fb49.PNG
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > <
> > https://github.com/zestedesavoir/zds-site/pull/3331#issuecomment-177955834
> > 
> > .
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/zestedesavoir/zds-site/pull/3331#issuecomment-177956901
> .

firm1 commented:
Autre souci.

J'ai désinscris l'ancien utilisateur de [ce tutoriel](http://zds.francoisdambrine.me/tutoriels/29/fugit-aut-voluptas-qui-sed/). Le tutoriel est bien allé à external, mais quand je vais sur le lien du tutoriel, je ne vois pas `external` comme auteur.

firm1 commented:
J'ai aussi une erreur 500 sur ce lien (connecté via admin) : http://zds.francoisdambrine.me/tutoriels/?tag=molestiae

artragis commented:
attend que j'ai corrigé le problème du front avant de faire des gros tests
car parfois un fichier mal placé peut générer des erreurs.

2016-02-01 14:19 GMT+01:00 firm1 notifications@github.com:

> J'ai aussi une erreur 500 sur ce lien :
> http://zds.francoisdambrine.me/tutoriels/?tag=molestiae
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/zestedesavoir/zds-site/pull/3331#issuecomment-177968818
> .

artragis commented:
C'est bon de mon côté.

Le 1 février 2016 à 14:40, francois dambrine artragis@gmail.com a écrit :

> attend que j'ai corrigé le problème du front avant de faire des gros tests
> car parfois un fichier mal placé peut générer des erreurs.
> 
> 2016-02-01 14:19 GMT+01:00 firm1 notifications@github.com:
> 
> > J'ai aussi une erreur 500 sur ce lien :
> > http://zds.francoisdambrine.me/tutoriels/?tag=molestiae
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/zestedesavoir/zds-site/pull/3331#issuecomment-177968818
> > .

firm1 commented:
J'ai encore l'erreur 500 cela dit

artragis commented:
``` bash
[01/Feb/2016 14:51:58] "GET /tutoriels/?tag=molestiae HTTP/1.0" 500 126796
Internal Server Error: /tutoriels/
Traceback (most recent call last):
  File
"/home/francois/zds-site/zds-env/local/lib/python2.7/site-packages/django/core/handlers/base.py",
line 132, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File
"/home/francois/zds-site/zds-env/local/lib/python2.7/site-packages/django/views/generic/base.py",
line 71, in view
    return self.dispatch(request, *args, **kwargs)
  File
"/home/francois/zds-site/zds-env/local/lib/python2.7/site-packages/django/views/generic/base.py",
line 89, in dispatch
    return handler(request, *args, **kwargs)
  File
"/home/francois/zds-site/zds-env/local/lib/python2.7/site-packages/django/views/generic/list.py",
line 159, in get
    self.object_list = self.get_queryset()
  File "/home/francois/zds-site/zds/tutorialv2/views/views_published.py",
line 299, in get_queryset
    self.tag = get_object_or_404(SubCategory,
slug=self.request.GET.get('tag'))
  File
"/home/francois/zds-site/zds-env/local/lib/python2.7/site-packages/django/shortcuts.py",
line 155, in get_object_or_404
    return queryset.get(*args, **kwargs)
  File
"/home/francois/zds-site/zds-env/local/lib/python2.7/site-packages/django/db/models/query.py",
line 338, in get
    (self.model._meta.object_name, num)
MultipleObjectsReturned: get() returned more than one SubCategory -- it
returned 2!
[01/Feb/2016 14:51:58] "GET /tutoriels/?tag=molestiae HTTP/1.0" 500 126796
Internal Server Error: /tutoriels/
Traceback (most recent call last):
  File
"/home/francois/zds-site/zds-env/local/lib/python2.7/site-packages/django/core/handlers/base.py",
line 132, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File
"/home/francois/zds-site/zds-env/local/lib/python2.7/site-packages/django/views/generic/base.py",
line 71, in view
    return self.dispatch(request, *args, **kwargs)
  File
"/home/francois/zds-site/zds-env/local/lib/python2.7/site-packages/django/views/generic/base.py",
line 89, in dispatch
    return handler(request, *args, **kwargs)
  File
"/home/francois/zds-site/zds-env/local/lib/python2.7/site-packages/django/views/generic/list.py",
line 159, in get
    self.object_list = self.get_queryset()
  File "/home/francois/zds-site/zds/tutorialv2/views/views_published.py",
line 299, in get_queryset
    self.tag = get_object_or_404(SubCategory,
slug=self.request.GET.get('tag'))
  File
"/home/francois/zds-site/zds-env/local/lib/python2.7/site-packages/django/shortcuts.py",
line 155, in get_object_or_404
    return queryset.get(*args, **kwargs)
  File
"/home/francois/zds-site/zds-env/local/lib/python2.7/site-packages/django/db/models/query.py",
line 338, in get
    (self.model._meta.object_name, num)
MultipleObjectsReturned: get() returned more than one SubCategory -- it
returned 2!

```

2016-02-01 14:53 GMT+01:00 firm1 notifications@github.com:

> J'ai encore l'erreur 500 cela dit
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/zestedesavoir/zds-site/pull/3331#issuecomment-177979962
> .

firm1 commented:
Connecté en tant qu'admin, j'ai une erreur 500 sur le [profile anonyme](http://zds.francoisdambrine.me/membres/voir/anonymous/) (sur tous les profils en fait)

``` http
Request Method:     GET
Request URL:    http://zds.francoisdambrine.me/membres/voir/user/
Django Version:     1.8.8
Exception Type:     NoReverseMatch
Exception Value:    

Reverse for 'membe-settings_promote' with arguments '(3L,)' and keyword arguments '{}' not found. 0 pattern(s) tried: []

Exception Location:     /home/francois/zds-site/zds-env/local/lib/python2.7/site-packages/django/core/urlresolvers.py in _reverse_with_prefix, line 495
Python Executable:  /home/francois/zds-site/zds-env/bin/python
Python Version:     2.7.9
Python Path:    

['/home/francois/zds-site/zds-env/local/lib/python2.7/site-packages/git/ext/gitdb',
 '/home/francois/zds-site',
 '/home/francois/zds-site/zds-env/lib/python2.7',
 '/home/francois/zds-site/zds-env/lib/python2.7/plat-i386-linux-gnu',
 '/home/francois/zds-site/zds-env/lib/python2.7/lib-tk',
 '/home/francois/zds-site/zds-env/lib/python2.7/lib-old',
 '/home/francois/zds-site/zds-env/lib/python2.7/lib-dynload',
 '/usr/lib/python2.7',
 '/usr/lib/python2.7/plat-i386-linux-gnu',
 '/usr/lib/python2.7/lib-tk',
 '/home/francois/zds-site/zds-env/local/lib/python2.7/site-packages',
 '/home/francois/zds-site/zds-env/local/lib/python2.7/site-packages/gitdb/ext/smmap']

Server time:    lun, 1 Fév 2016 15:04:00 +0100
```

artragis commented:
> Autre souci.
> 
> J'ai désinscris l'ancien utilisateur de ce tutoriel. Le tutoriel est bien allé à external, mais quand je vais sur le lien du tutoriel, je ne vois pas external comme auteur.

c'est un bug déjà présent en prod, ce n'est pas la faute de django1.8. C'est juste que la désinscription ne remplace pas l'historisation des auteurs (une PR récente).

artragis commented:
> J'ai aussi une erreur 500 sur ce lien (connecté via admin) : http://zds.francoisdambrine.me/tutoriels/?tag=molestiae

c'est un bug présent en prod mais qu'il convient de régler : il semblerait qu'aucune contrainte d'unicité n'existe sur les catégories/tags. Autrement dit on est sûrement en train de se foirer nous même nos perfs à cause de ça. Il faut donc ajouter cette contrainte là.

gustavi commented:
Ça sera sûrement corrigé dans la ZEP-25

artragis commented:
mais comme la v17 avec django 1.8 (ou 1.9 si les dépendances répondent à
temps) arrivera nécessairement avant la zep-25 il faut un fix, même
temporaire et subobtimal.

2016-02-01 16:30 GMT+01:00 Laville Augustin notifications@github.com:

> Ça sera sûrement corrigé dans la ZEP-25
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/zestedesavoir/zds-site/pull/3331#issuecomment-178021178
> .

gustavi commented:
https://www.djangoproject.com/weblog/2016/feb/01/releases-192-and-189/

SpaceFox commented:
« Django 1.7 and older are no longer receiving security updates but are
also unaffected. » ?

2016-02-01 19:08 GMT+01:00 Laville Augustin notifications@github.com:

> https://www.djangoproject.com/weblog/2016/feb/01/releases-192-and-189/
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/zestedesavoir/zds-site/pull/3331#issuecomment-178103757
> .

gustavi commented:
C'était juste pour qu'on pense à changer la version.

gustavi commented:
@GerardPaligot tu peux passer Django en 1.8.9 ?

GerardPaligot commented:
Done et j'ai rollback aussi la documentation pour Django 1.8.

DevHugo commented:
La page [aprops](https://zestedesavoir.com/pages/apropos/) à t-elle été modifiée ?

GerardPaligot commented:
@DevHugo Bien vu !

gustavi commented:
On peut en profiter pour mettre à jour la version de Pillow (classé _insecure_ actuellement) ?

GerardPaligot committed:
chore(dep): Mise a jour de pillow de 2.9 a 3.1.1.

GerardPaligot commented:
Pillow mis à jour.

GerardPaligot commented:
Je rappel que la release v17 est consacrée uniquement à cette PR. Nous avons grand besoin de QA pour continuer à développer parce que rien ne sera mergée avant que cette PR ne le soit et qu'elle soit incluse dans une release.

artragis commented:
un passage par la preprod?
sinon demain je clean mon serveur (la bdd est instable) et je tire ta PR.

Le 17/02/2016 21:42, Gérard Paligot a écrit :

> Je rappel que la release v17 est consacrée uniquement à cette PR. Nous 
> avons grand besoin de QA pour continuer à développer parce que rien ne 
> sera mergée avant que cette PR ne le soit et qu'elle soit incluse dans 
> une release.
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/zestedesavoir/zds-site/pull/3331#issuecomment-185395249.

GerardPaligot commented on zds/utils/models.py:
```diff
@@ -131,7 +131,7 @@ class Meta:
                                related_name='comments', db_index=True)
     editor = models.ForeignKey(User, verbose_name='Editeur',
-                               related_name='comments-editor',
+                               related_name='comments-editor+',
```
@gustavi Tu sais me dire pourquoi tu as rajouté ce "+" ?

vhf replied:
https://docs.djangoproject.com/fr/1.9/ref/models/fields/#django.db.models.ForeignKey.related_name ;)

SpaceFox commented:
J'avoue ne pas bien comprendre pourquoi du coup, on a des commits 1.7 --> 1.8, des commits 1.8 --> 1.9, puis des commits  1.9 --> 1.8. En toute logique, on ne devrait avoir que la première série.

GerardPaligot commented:
A la base, @gustavi voulait sauter la version 1.8 pour passer directement à la version 1.9 mais nos dépendances ne sont pas prêtes. Par contre, son travail (avec la collaboration de @DevHugo il me semble) résolvait ton problème de l'époque.

Je suis donc parti de sa PR. Cela dit, je peux rebase le tout le plus proprement possible si tu le souhaites.

SpaceFox commented:
J'ai pas étudié l'historique des commits, mais en théorie on devrait s'en sortir en ne prenant que la première partie de la branche non ? Éventuellement avec quelques cherry-picks. Parce qu'en l'état, l'historique des commits est juste incompréhensible.

J'en profite pour penser que c'est l'occasion de [faire un tour dans cette liste](https://requires.io/github/zestedesavoir/zds-site/requirements/?branch=dev), quitte à se taper une version technique et chiante à QA, autant y aller à fond.

GerardPaligot commented:
@SpaceFox Justement, les 3/4 des commits se sont des mises à jour de nos dépendances et pas réellement une migration vers django 1.9 (comme en témoigne mon commit de rollback ridiculement petit en modification).

gustavi commented:
Effectivement, notre code est compatible déjà compatible 1.9 mais fonctionne sous 1.8. Ça évite d'avoir du travail supplémentaire pour la 1.9.

artragis commented:
> J'ai aussi une erreur 500 sur ce lien (connecté via admin) : http://zds.francoisdambrine.me/tutoriels/?tag=molestiae

J'ai supprimé l'erreur 500 par contre faudra absolument modifier notre modèle pour que le slug devienne unique svp (mettre un `index=True` dans utils.models.subcategory il me semble). 

TOut est testable sur mon serveur.

GerardPaligot commented:
Sachez que mon rebase est toujours en cours mais c'est un peu long. Il y a des commits d'un peu tout après ceux de SpaceFox.

@artragis J'ai déjà corrigé 2-3 bugs, tu sais retirer ma branche ?

artragis commented:
ce midi.

Le 18 février 2016 à 09:42, Gérard Paligot notifications@github.com a
écrit :

> Sachez que mon rebase est toujours en cours mais c'est un peu long. Il y a
> des commits d'un peu tout après ceux de SpaceFox.
> 
> @artragis https://github.com/artragis J'ai déjà corrigé 2-3 bugs, tu
> sais retirer ma branche ?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/zestedesavoir/zds-site/pull/3331#issuecomment-185599298
> .

artragis commented:
c'est bon/

Au passage j'ai un petit warning pas méchant sur 

```
/home/francois/zds-site/zds-env/local/lib/python2.7/site-packages/corsheaders/middleware.py:10: RemovedInDjango19Warning: The utilities in django.db.models.loading are deprecated in favor of the new application loading system.
  from django.db.models.loading import get_model
```

artragis commented on requirements.txt:
```diff
@@ -4,17 +4,17 @@ pygments==2.0.2
 # Explicit dependencies (references in code)
-django==1.7.10
+Django==1.8.9
 coverage==4.0.1
```
version 4.0.3, à mettre sûrement dans requirements-dev non?

artragis commented on requirements.txt:
```diff
@@ -23,15 +23,15 @@ beautifulsoup4==4.3.2
 django-recaptcha==1.0.4
```
bs4 est à passer en version 4.4.1
cairosvg en 1.0.19

GerardPaligot replied:
Done.

artragis commented on requirements.txt:
```diff
@@ -4,17 +4,17 @@ pygments==2.0.2
 pygeoip==0.3.2
-pillow==2.9.0
+pillow==3.1.1
 gitpython==1.0.1
```
version 1.0.2 (il semble même que c'est un de nos commit qui est en jeu sur cette dep)

firm1 replied:
attention à la version 1.0.2 de gitpython, j'ai vu passer des regression sur leur dépot. Je regarderai ça de plus près pour voir en quoi c'est impactant

GerardPaligot replied:
On va éviter pour le moment alors.

artragis commented on requirements.txt:
```diff
@@ -4,17 +4,17 @@ pygments==2.0.2
+django-munin==0.2.1
 python-memcached==1.54
 lxml==3.4.4
 factory-boy==2.4.1
```
version 2.6.1

artragis commented on requirements.txt:
```diff
@@ -23,15 +23,15 @@ beautifulsoup4==4.3.2
 
 # Zep 12 dependency
-django-uuslug==1.0.3
+django-uuslug==1.1.6
```
A tester fortement, cette version opère des changements sur l'algo de slug.
si ça marche pas rollback 1.1.5
Si ça marche up to 1.1.7

artragis commented on requirements.txt:
```diff
@@ -23,15 +23,15 @@ beautifulsoup4==4.3.2
 # Api dependencies
-djangorestframework==3.1.1
+djangorestframework==3.3.2
 djangorestframework-xml==1.0.1
```
la version 1.3.0 existe mais je n'ai aucun retour sur le changelog.

GerardPaligot committed:
chore(dep): Mise a jour de django-filter de 0.9.2 a 0.12.

GerardPaligot committed:
chore(dep): Mise a jour de django_model_utils de 2.4 a 2.4.

GerardPaligot committed:
chore(dep): Mise a jour de django-rest-swagger de 0.2.9 a 0.3..0.

GerardPaligot committed:
chore(dep): Mise a jour de DRF de 3.3.1 a 3.3.2.

Mise a jour de dependances connexes :
- drf-extensions de 0.2.7 a 0.2.8.
- django-cors-middleware de 1.0.0 a 1.2.0.
- Suppression de get_cache()
- Changement de noms de parametres de configuration pour REST_FRAMEWORK

Merci particulier a @DevHugo pour sa contribution.

GerardPaligot commented:
Bon, je pense qu'on commence à arriver à un historique un peu près propre (le reste est chiant pour pas grand chose) mais subitemment, la branche s'est mise à ne plus passer alors que je n'ai eu aucun problème pendant le rebase.

Si quelqu'un à une idée, PR welcome. D'après mes recherches et les diffs de ma branche, rien n'explique cet échec.

GerardPaligot commented:
Vu le build Travis suite au rapatriment d'un commit de la release v16 vers dev, l'erreur sur cette PR n'est pas liée à la PR mais au dépôt. Plus d'info [ici](https://github.com/zestedesavoir/zds-site/issues/3381).

pierre-24 commented:
Je relance: c'est un problème côté téléchargement de TeX, mais les mirroirs CTAN ont l'air de nouveaux utilisables :)

artragis commented:
yay !

Plus qu'à rebase pour les conflits :p

GerardPaligot commented:
Rebased mais je me demande si je ne ferais pas une nouvelle PR parce que l'ordre dans les commits est toujours foireux (il en skip pas mal là pour travis en plus).

GerardPaligot committed:
chore(dep): Mise a jour de coverage de 4.0.1 a 4.0.3.

GerardPaligot committed:
chore(dep): Mise a jour de cairosvg de 1.0.13 a 1.0.19.

GerardPaligot committed:
chore(dep): Mise a jour de beautifulsoup4 de 4.3.2 a 4.4.1.

GerardPaligot committed:
chore(dep): Mise a jour de djangorestframework-xml de 1.0.1 a 1.3.0.

GerardPaligot committed:
chore(dep): Mise a jour de factory-boy de 2.4.1 a 2.6.1.

GerardPaligot commented:
Avis à tous les contributeurs, cette PR est prête pour la QA et est nécessaire pour commencer à merger nos autres PRs.

Comme cette PR touche à absolument tout le projet, nous devons faire une QA locale minimale et une QA importante sur la bêta . Si vous ne disposez pas d'une instance zds locale, @artragis propose son serveur avec la branche ci-présente tirée : http://zds.francoisdambrine.me/ (merci à lui).

Je vous encourage donc tous à faire cette QA importante ! :)

SpaceFox commented:
« chore(dep) » dans le sens « travail routinier » ?

GerardPaligot commented:
@zestedesavoir/qa Il nous faut ici une QA pour merger, faire une release et pouvoir commencer à merger nos PRs en attente ! :)

@SpaceFox https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#type :)

gustavi commented:
Problème : je n'arrive pas à envoyer un mp à `test` via `admin` (`Group matching query does not exist.`). Simple problème au niveau de cet instance ou véritable bug ?

gustavi commented:
OK au temps pour moi ça vient de 

```
excluded_groups = [Group.objects.get(name=settings.ZDS_APP['member']['bot_group'])]
```

artragis commented:
Du coup c'est une _sorte de bug_ de mon instance. En fait l'identifiant 
de l'utilisateur test est celui que devrait avoir l'utilisateur bot mais 
comme j'ai chargé les fixtures de base après avoir lancé la commande qui 
a tout chargé, bah ça se peut que ça ait fait un petit conflit de rien 
du tout.

Le 20/02/2016 11:41, Laville Augustin a écrit :

> Problème : je n'arrive pas à envoyer un mp à |test| via |admin| 
> (|Group matching query does not exist.|). Simple problème au niveau de 
> cet instance ou véritable bug ?
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/zestedesavoir/zds-site/pull/3331#issuecomment-186566606.

Emeric54 commented:
Je continue de jouer avec tout ce qui peut-être testé, pour l'instant, RAS.

gustavi commented:
**QA** : RAS. Pour moi c'est good, j'ai testé un max de trucs et j'ai rien vu d'anormal.

Juste : 
- On a pas de recherche sur cette instance ;
- Je n'ai pas testé l'API ;
- Je n'ai pas testé la désinscription.

artragis commented:
non je n'ai aps mis la recherche,
non je n'ai aps mis la génération des exports de contenu,
non je n'ai pas mis le captcha

Le 20/02/2016 12:08, Laville Augustin a écrit :

> _QA_ : RAS. Pour moi c'est good, j'ai testé un max de trucs et j'ai 
> rien vu d'anormal.
> 
> Juste :
> - On a pas de recherche sur cette instance ;
> - Je n'ai pas testé l'API ;
> - Je n'ai pas testé la désinscription.
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/zestedesavoir/zds-site/pull/3331#issuecomment-186573081.

pierre-24 commented:
Je crois me souvenir que la recherche, c'est important de tester, non ?

artragis commented:
Mais sans le contenu réel de la prod, ça sert à rien, on va perdre du 
temps pour rien.

Le 20/02/2016 12:09, Pierre Beaujean a écrit :

> Je crois me souvenir que la recherche, c'est important de tester, non ?
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/zestedesavoir/zds-site/pull/3331#issuecomment-186573547.

gustavi commented:
Oui il faut tester la recherche mais on fera ça sur la beta. Perso j'ai pas d'env avec la recherche de propre sous la main.

GerardPaligot commented:
Merci tout le monde ! Alors, si c'est ok pour @SpaceFox, on merge et on lance la release (dès qu'il aura du temps pour la faire du coup).

SpaceFox commented:
Dimanche soir, donc 😃
Le 20 févr. 2016 13:31, "Gérard Paligot" notifications@github.com a
écrit :

> Merci tout le monde ! Alors, si c'est ok pour @SpaceFox
> https://github.com/SpaceFox, on merge et on lance la release (dès qu'il
> aura du temps pour la faire du coup).
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/zestedesavoir/zds-site/pull/3331#issuecomment-186584339
> .

firm1 commented:
En testant **très rapidement** sur le serveur de @artragis voici quelques remarques (j'ignore si ça vient des _fixtures_ mal chargées) : 
- Erreur 500 sur [cette page](http://zds.francoisdambrine.me/membres/)
- Les extras (pdf, html, epub) ne sont pas générés pour [cet article](http://zds.francoisdambrine.me/articles/64/la-france-et-les-francais/) que je viens de publier

gustavi commented:
C'est "normal" @firm1 pour la 500, c'est parce que la configuration n'est pas faite à 100% (pas de groupe pour les _bot_ ce qui cause ce souci). Pour l'autre j'avoue que je sèche.

Ping @artragis @sandhose la config du serveur est OK à ce niveau ? Ça peut venir de quoi ?

firm1 commented:
@gustavi : du coup le problème des groupes m'empeche de tester pas mal
d'urls.

Je vais peut etre attendre la mise en beta pour tester correctement.

Le lun. 22 févr. 2016 10:41, Laville Augustin notifications@github.com a
écrit :

> C'est "normal" @firm1 https://github.com/firm1 pour la 500, c'est parce
> que la configuration n'est pas faite à 100% (pas de groupe pour les _bot_
> ce qui cause ce souci). Pour l'autre j'avoue que je sèche.
> 
> Ping @artragis https://github.com/artragis @sandhose
> https://github.com/sandhose la config du serveur est OK à ce niveau ?
> Ça peut venir de quoi ?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/zestedesavoir/zds-site/pull/3331#issuecomment-187096490
> .

artragis commented:
Je fais un reset bdd + un truc clean dans 15 minutes

Le 22 février 2016 à 11:28, firm1 notifications@github.com a écrit :

> @gustavi : du coup le problème des groupes m'empeche de tester pas mal
> d'urls.
> 
> Je vais peut etre attendre la mise en beta pour tester correctement.
> 
> Le lun. 22 févr. 2016 10:41, Laville Augustin notifications@github.com a
> écrit :
> 
> > C'est "normal" @firm1 https://github.com/firm1 pour la 500, c'est
> > parce
> > que la configuration n'est pas faite à 100% (pas de groupe pour les _bot_
> > ce qui cause ce souci). Pour l'autre j'avoue que je sèche.
> > 
> > Ping @artragis https://github.com/artragis @sandhose
> > https://github.com/sandhose la config du serveur est OK à ce niveau ?
> > Ça peut venir de quoi ?
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > <
> > https://github.com/zestedesavoir/zds-site/pull/3331#issuecomment-187096490
> > 
> > .
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/zestedesavoir/zds-site/pull/3331#issuecomment-187113834
> .

artragis commented:
Voilà tout est OK

Le 22 février 2016 à 12:15, francois dambrine artragis@gmail.com a écrit :

> Je fais un reset bdd + un truc clean dans 15 minutes
> 
> Le 22 février 2016 à 11:28, firm1 notifications@github.com a écrit :
> 
> > @gustavi : du coup le problème des groupes m'empeche de tester pas mal
> > d'urls.
> > 
> > Je vais peut etre attendre la mise en beta pour tester correctement.
> > 
> > Le lun. 22 févr. 2016 10:41, Laville Augustin notifications@github.com
> > a
> > écrit :
> > 
> > > C'est "normal" @firm1 https://github.com/firm1 pour la 500, c'est
> > > parce
> > > que la configuration n'est pas faite à 100% (pas de groupe pour les
> > > _bot_
> > > ce qui cause ce souci). Pour l'autre j'avoue que je sèche.
> > > 
> > > Ping @artragis https://github.com/artragis @sandhose
> > > https://github.com/sandhose la config du serveur est OK à ce niveau ?
> > > Ça peut venir de quoi ?
> > > 
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <
> > > https://github.com/zestedesavoir/zds-site/pull/3331#issuecomment-187096490
> > > 
> > > .
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/zestedesavoir/zds-site/pull/3331#issuecomment-187113834
> > .

firm1 commented:
Après le clean, 
- la 500 sur les membres à disparue : cool.
- le bug des extras non généré est toujours là lui : pas cool
- erreur 500 si j'essaye de mettre une LS temporaire sans renseigner la durée de la LS dans la modale : pas cool

artragis commented:
> le bug des extras non généré est toujours là lui : pas cool

il le restera : c'est pas un bug, j'ai configuré les extras sur "nothing" c'est déjà dit.

> rreur 500 si j'essaye de mettre une LS temporaire sans renseigner la durée de la LS dans la modale : pas cool

là par contre on a un bug.

artragis commented:
Pour le bug, je n'y arrive pas cher firm1
Quel membres as-tu tenter de lectureseuler?

firm1 commented:
En tant qu'admin, j'ai tenté de mettre une LS à "userstaff9"

artragis commented:
j'y arrive toujours pas : avais-tu rempli tous les champs?

Le 22 février 2016 à 14:05, firm1 notifications@github.com a écrit :

> En tant qu'admin, j'ai tenté de mettre une LS à "userstaff9"
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/zestedesavoir/zds-site/pull/3331#issuecomment-187162462
> .

sandhose commented:
c'est la peine que je fasse un dump de la beta vers mon serveur, ou cette PR va bientôt être déployée en beta ? (je peux d'ailleurs le faire, s'il le faut)

SpaceFox commented:
Vu qu'apparemment les problèmes rencontrés tiennent de la merdouille et qu'on ne peut pas tester plein de trucs sans passer par la bêta, il faudrait lancer rapidement la v17-RC1 :
1. Merger cette PR
2. Créer la v17-RC1 (tag + branche de release)
3. Déploiement sur la bêta
4. Créer le topic qui va bien sur les forums.

sandhose commented on scripts/deploy.sh:
```diff
@@ -31,7 +31,7 @@ cd /opt/zdsenv/ZesteDeSavoir/
 sudo rm /etc/nginx/sites-enabled/zestedesavoir
 sudo ln -s /etc/nginx/sites-available/zds-maintenance /etc/nginx/sites-enabled/zds-maintenance
-sudo systemctl reload nginx.service
+sudo service nginx reload
```
Euh, pourquoi ?

Même si _techniquement_, ça marche, puisque `systemd` ship avec une couche de compat SysVinit, je comprend pas cette modif, sachant qu'il a jamais été question d'être compatible avec tous les systèmes, dont ceux qui sont pas sous systemd.

SpaceFox replied:
Je ne sais pas le pourquoi, mais cette modif n'a pas lieu d'être. À supprimer.

GerardPaligot replied:
@SpaceFox Cette modification a été faite dans l'un de tes commits : https://github.com/GerardPaligot/zds-site/commit/8b2ddae1f9562670edfc5ea52f78b20744649d20

SpaceFox replied:
[C'est visiblement une erreur de rebase](https://github.com/GerardPaligot/zds-site/commit/fb774b8a3ca1c40b878afd6c6c50dbd993d2d424).

GerardPaligot commented:
Notez quand même l'urgence d'une autre issue : https://github.com/zestedesavoir/zds-site/issues/3383

GerardPaligot commented:
Mise à jour de la branche pour retirer la mise à jour de `uuslug`. :)

pierre-24 commented:
Moi j'ai hotfixé, ça me semble nécessaire.

SpaceFox commented:
Excellente initiative @pierre-24 !

gustavi commented:
Du coup il nous reste quoi ici ?

SpaceFox commented:
Le hotfix.
Le 23 févr. 2016 19:24, "Laville Augustin" notifications@github.com a
écrit :

> Du coup il nous reste quoi ici ?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/zestedesavoir/zds-site/pull/3331#issuecomment-187829111
> .

SpaceFox commented:
Haha… c'est en conflit :(

gustavi commented:
\o/
